### PR TITLE
Document ThreadSanitizer in unstable-book

### PR DESCRIPTION
**Pull request by tmiasko**

### CODE REVIEW

**Comment by lzutao** [comment]
Just curious: Why don't you add committer email address to your github account?

File: src/doc/unstable-book/src/compiler-flags/sanitizer.md [hunk 1]
```diff
@@ -164,17 +164,25 @@ Shadow byte legend (one shadow byte represents 8 application bytes):
...
   Shadow gap:              cc
-==424427==ABORTING
+==39249==ABORTING
 ```
 
-## MemorySanitizer
+# MemorySanitizer
+
+MemorySanitizer is detector of uninitialized reads. It is supported only on
+`x86_64-unknown-linux-gnu` target.
```
Comment (steveklabnik) [inline src/doc/unstable-book/src/compiler-flags/sanitizer.md]: This second sentence should be slightly re-worded:

> It is only supported on the `x86_64-unknown-linux-gnu` target.

File: src/doc/unstable-book/src/compiler-flags/sanitizer.md [hunk 2]
```diff
@@ -164,17 +164,25 @@ Shadow byte legend (one shadow byte represents 8 application bytes):
...
+
+MemorySanitizer requires all program code to be instrumented. Failing to achieve
+that will result in false positive reports.
```
Comment (steveklabnik) [inline src/doc/unstable-book/src/compiler-flags/sanitizer.md]: This means "if you link to C code it needs to be compiled with memorysantizer too", right?

**Review by steveklabnik — CHANGES_REQUESTED** [review]
Thanks for this! Two minor nits. I also second the other comment; you may want to change the email you've committed with.

File: src/doc/unstable-book/src/compiler-flags/sanitizer.md (hunk 2 above)
Comment (tmiasko) [reply src/doc/unstable-book/src/compiler-flags/sanitizer.md]: Yes, expanded on this a bit to give more context to the example that follows.

In general this is a major challenge when using MemorySanitizer in larger code
bases. For those written mostly in Rust and using Cargo, it works surprisingly
well.

**Review by steveklabnik — APPROVED** [review]
Thanks!

**Pull request merged**